src/main/java/uk/gov/ida/matchingserviceadapter/saml/UserIdHashFactory.java

```diff
@@ -4,47 +4,51 @@
 import org.apache.commons.codec.binary.Hex;
 import org.opensaml.security.crypto.JCAConstants;
 import uk.gov.ida.saml.core.domain.AuthnContext;
-import uk.gov.ida.saml.core.domain.IdentityProviderAuthnStatement;
 
 import java.io.UnsupportedEncodingException;
 import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
 import java.text.MessageFormat;
 
-import static com.google.common.base.Throwables.propagate;
-
 public class UserIdHashFactory {
-    public String createHashedId(String partnerEntityId, String entityId, String persistentId, Optional<IdentityProviderAuthnStatement> authnStatement) {
+
+    private final String msaEntityId;
+
+    public UserIdHashFactory(String msaEntityId) {
+        this.msaEntityId = msaEntityId;
+    }
+
+    public String hashId(String issuerEntityId, String persistentId, Optional<AuthnContext> authnContext) {
         MessageDigest md;
         try {
             md = MessageDigest.getInstance(JCAConstants.DIGEST_SHA256);
         } catch (NoSuchAlgorithmException e) {
-            throw propagate(e);
+            throw new RuntimeException(e);
```

Comment:
I don't think we should just throw generic RunTimeException. If we have issues generating the Hash Id it's a pretty serious deal so we should probably have a custom exception for that. (Although it will still be unchecked exception...)